qutip/solver/mcsolve.py

```diff
@@ -561,6 +598,158 @@ def run(
         result.stats['run time'] = time() - start_time
         return result
 
+    def run_mixed(
```

Comment:
Question: Our currently solver classes have a single `run` method (with some exceptions). It would be nice if we could maintain that here. Is it possible to have a `run` method that calls either `run_mixed` or `run_pure`? Or two have two solvers or combine the two methods? Not sure if any of these options are sensible.

Reply:
I was waiting for this review comment since I wasn't sure how to go about it :) Yes, it would be possible to have a single `run` method. However, some arguments have different types for run_mixed, and run_mixed does not accept the `target_tol` parameter. I thought that might be a bit confusing, but if you think it is better, I will change it to a single method (which would call either `_run_pure` or `_run_mixed` or so).

Reply:
Calling `_run_pure` and `_run_mixed` sounds good. I guess it depends how different the arguments are. One parameter not accepted doesn't seem like the end of the world. What are the different types?

Reply:
Thanks for making the change. The big picture structure looks good to me now. I'll leave the detailed review to @Ericgig since he knows `mcsolve` much better than I do.